lib/rets/metadata/file_cache.rb

```diff
@@ -0,0 +1,29 @@
+module Rets
+  module Metadata
+
+    # This metadata cache persists the metadata to a file.
+    class FileCache
+
+      def initialize(path)
+        @path = path
+      end
+
+      # Save the metadata.
+      def save(metadata)
+        File.open(@path, "wb") do |file|
+          Marshal.dump(metadata, file)
+        end
+      end
+
+      # Load the metadata.  Return nil if it could not be loaded.
+      def load
+        File.open(@path, "rb") do |file|
+          Marshal.load(file)
+        end
+      rescue IOError, SystemCallError
+        nil
+      end
```

Comment:
@hfaults I'm not sure how the problem of having to set the logger via a mutator is a result of using Marshal... it's more a result of my choice to serialize the root object rather than sources.  We'd have the same problem if serializing the root object with JSON.  If that mutator is a source of irritation, I don't see any big problem with serializing the source instead.

@tcrayford It makes sense for the user to be able to use whatever serialization method they like.  I can think of three that the library could support:

* Marshall
* YAML
* JSON

Are there any others?

By "support," I mean that the root object should be able to be serialized.  We pass the root metadata object to the cache object, and the cache object uses whichever serialization method that the user prefers.